pkg/policy/api/cidr.go

```diff
@@ -75,19 +67,24 @@ func (s CIDRSlice) GetAsEndpointSelectors() EndpointSelectorSlice {
 	// If multiple CIDRs representing reserved:world are in this CIDRSlice,
 	// we only have to add the EndpointSelector representing reserved:world
 	// once.
-	var hasWorldBeenAdded bool
+	var hasIPv4AllBeenAdded, hasIPv6AllBeenAdded bool
 	slice := EndpointSelectorSlice{}
 	for _, cidr := range s {
-		if cidr.MatchesAll() && !hasWorldBeenAdded {
-			hasWorldBeenAdded = true
-			slice = append(slice, ReservedEndpointSelectors[labels.IDNameWorld])
+		if cidr == ipv4All {
+			hasIPv4AllBeenAdded = true
+		}
+		if cidr == ipv6All {
+			hasIPv6AllBeenAdded = true
 		}
 		lbl, err := cidrpkg.IPStringToLabel(string(cidr))
 		if err == nil {
 			slice = append(slice, NewESFromLabels(lbl))
 		}
 		// TODO: Log the error?
 	}
+	if hasIPv4AllBeenAdded && hasIPv6AllBeenAdded {
```

Comment:
Couldn't we take into account whether we are running an IPv4-only or IPv6-only cluster? As Joe mentioned offline, this fix may have a big impact on the policy map size because that one entry will now be converted into one entry per existing CIDR IIUC. So if we can limit its effect to dual-stack clusters, that would probably be best.